test/test_chapeldomain.py

```diff
@@ -766,51 +767,56 @@ def test_with_prefixes(self):
         ]
         for sig, prefix, name, arglist in test_cases:
-            self.check_sig(sig, prefix, None, name, arglist, None)
+            self.check_sig(sig, prefix, None, name, arglist, None, None)
 
     def test_with_all(self):
         """Verify fully specified signatures parse correctly."""
         test_cases = [
-            ('proc foo() ref', 'proc ', None, 'foo', '', ' ref'),
-            ('iter foo() ref', 'iter ', None, 'foo', '', ' ref'),
-            ('inline proc Vector.pop() ref', 'inline proc ', 'Vector.', 'pop', '', ' ref'),
-            ('inline proc range.first', 'inline proc ', 'range.', 'first', None, None),
-            ('iter Math.fib(n: int(64)): GMP.BigInt', 'iter ', 'Math.', 'fib', 'n: int(64)', ': GMP.BigInt'),
-            ('proc My.Mod.With.Deep.NameSpace.1.2.3.432.foo()', 'proc ', 'My.Mod.With.Deep.NameSpace.1.2.3.432.', 'foo', '', None),
-            ('these() ref', None, None, 'these', '', ' ref'),
-            ('size', None, None, 'size', None, None),
-            ('proc Util.toVector(type eltType, cap=4, offset=0): Containers.Vector', 'proc ', 'Util.', 'toVector', 'type eltType, cap=4, offset=0', ': Containers.Vector'),
-            ('proc MyClass$.lock$(combo$): sync bool', 'proc ', 'MyClass$.', 'lock$', 'combo$', ': sync bool'),
-            ('proc MyClass$.lock$(combo$): sync myBool$', 'proc ', 'MyClass$.', 'lock$', 'combo$', ': sync myBool$'),
-            ('proc type currentTime(): int(64)', 'proc type ', None, 'currentTime', '', ': int(64)'),
-            ('proc param int.someNum(): int(64)', 'proc param ', 'int.', 'someNum', '', ': int(64)'),
-            ('proc MyRs(seed: int(64)): int(64)', 'proc ', None, 'MyRs', 'seed: int(64)', ': int(64)'),
+            ('proc foo where a > b', 'proc ', None, 'foo', None, None, ' where a > b'), 
+            ('proc foo() where a > b', 'proc ', None, 'foo', '', None, ' where a > b'), 
+            ('proc foo:int where a > b', 'proc ', None, 'foo', None, ':int', ' where a > b'), 
+            ('proc foo():int where a > b', 'proc ', None, 'foo', '', ':int', ' where a > b'), 
+            ('proc foo ref where a > b', 'proc ', None, 'foo', None, ' ref', ' where a > b'), 
+            ('proc foo() ref where a > b', 'proc ', None, 'foo', '', ' ref', ' where a > b'), 
+            ('proc foo ref: int where a > b', 'proc ', None, 'foo', None, ' ref: int', ' where a > b'), 
+            ('proc foo() ref: int where a > b', 'proc ', None, 'foo', '', ' ref: int', ' where a > b'),
+            ('proc foo() ref', 'proc ', None, 'foo', '', ' ref', None),
+            ('iter foo() ref', 'iter ', None, 'foo', '', ' ref', None),
+            ('inline proc Vector.pop() ref', 'inline proc ', 'Vector.', 'pop', '', ' ref', None),
+            ('inline proc range.first', 'inline proc ', 'range.', 'first', None, None, None),
+            ('proc My.Mod.With.Deep.NameSpace.1.2.3.432.foo()', 'proc ', 'My.Mod.With.Deep.NameSpace.1.2.3.432.', 'foo', '', None, None),
```

Comment:
Looks like you dropped the Math.fib check that would go above this line, was that intentional?